src/views/CardStack/CardStackStyleInterpolator.js

```diff
@@ -49,17 +72,21 @@ function forHorizontal(
     return forInitial(props);
   }
 
-  const index = scene.index;
-  const inputRange = [index - 1, index, index + 1];
-
+  const scenes = formatScenes(props.scenes, props.index);
+  const lastIndex = scenes.length - 1;
+  const activeIndex = scenes.findIndex(
+    (item: NavigationScene) => item.isActive
+  );
+  const index = scenes.findIndex((item: NavigationScene) => item === scene);
+  const isBack = !scenes[lastIndex].isActive;
   const width = layout.initWidth;
+
+  const inputRange = [index - 1, index, index + 1];
   const outputRange = I18nManager.isRTL
     ? ([-width, 0, width * 0.3]: Array<number>)
     : ([width, 0, width * -0.3]: Array<number>);
 
-  // Add [index - 1, index - 0.99] to the interpolated opacity for screen transition.
-  // This makes the screen's shadow to disappear smoothly.
-  const opacity = position.interpolate({
+  let opacity = {
```

Comment:
To prevent reassignment or mutation of opacity, could the logic that determine's `opacity`'s value be its own function?

Reply:
Also I noticed that you removed the comments, was there motivation behind this?

Reply:
scenes: [A , B , C , D]
D -> A , 
B and C Does not need to show, so modified `opacity = {opacity: 0}`

Reply:
I just cancel the merge。